hooks/useContractorInterest.ts

```
@@ -0,0 +1,189 @@
+
+  const handlePageSizeChange = useCallback((size: number) => {
+    setPageSize(size);
+    setCurrentPage(1); // Reset to first page when changing page size
+  }, []);
+
+  const refreshContractors = useCallback(() => {
+    fetchContractors();
+  }, [fetchContractors]);
+
+  // CRUD operations
+  const createContractor = useCallback(async (data: any) => {
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)